src/monmove.cpp

```diff
@@ -277,10 +279,57 @@ float monster::rate_target( Creature &c, float best, bool smart ) const
     return INT_MAX;
 }
 
+bool monster::is_item_lootable( const item &itm )
+{
+    bool lootable = false;
+
+    //TODO: Definitely find a better way to do the below, too much repeatability
```

Comment:
Yes. How about a template function:

```C++
template<typename L, typename P>
static bool is_item_lootable( const item &itm, constL &list, const P &predicate, bool requires_all ) {
// Your loop goes here. Use list instead of lootable_categories, use predicate for the if-condition
}
// called like this:
if( !is_item_lootable( itm, lootable_categories, [](const item &itm, const std::string &tag) { return itm.get_category().name() == tag; }, lootables_requires_all ) {
return false;
}
```